package/gluon-mesh-batman-adv/Makefile

```diff
@@ -18,37 +18,37 @@ define Package/gluon-mesh-batman-adv/common
 	+libnl-tiny \
 	+libbatadv \
 	+batctl \
-	+@GLUON_SPECIALIZE_KERNEL:KERNEL_DUMMY \
-	+@GLUON_SPECIALIZE_KERNEL:KERNEL_CRC16 \
-	+@GLUON_SPECIALIZE_KERNEL:KERNEL_LIBCRC32C
+	+@GLUON_SPECIALIZE_KERNEL:KERNEL_DUMMY
 endef
 
-define Package/gluon-mesh-batman-adv-14
-$(Package/gluon-mesh-batman-adv/common)
-  TITLE:=Support for batman-adv meshing (compat level 14)
-  CONFLICTS:=gluon-mesh-batman-adv-15
-  DEPENDS+=+kmod-batman-adv-legacy
+define Package/gluon-mesh-batman-adv-legacy
+  TITLE:=Support for batman-adv-legacy meshing (compat level 14)
+  DEPENDS+= \
+	+gluon-mesh-batman-adv-common \
+	+kmod-batman-adv-legacy \
+	+@GLUON_SPECIALIZE_KERNEL:KERNEL_CRC16 \
+	+@GLUON_SPECIALIZE_KERNEL:KERNEL_LIBCRC32C
 endef
 
-define Package/gluon-mesh-batman-adv-15
-$(Package/gluon-mesh-batman-adv/common)
+define Package/gluon-mesh-batman-adv
   TITLE:=Support for batman-adv meshing (compat level 15)
-  DEPENDS+=+kmod-batman-adv
+  DEPENDS+= \
+	+gluon-mesh-batman-adv-common \
+	+kmod-batman-adv \
+	+@GLUON_SPECIALIZE_KERNEL:KERNEL_CRC16 \
+	+@GLUON_SPECIALIZE_KERNEL:KERNEL_LIBCRC32C
 endef
 
-define Package/gluon-mesh-batman-adv-14/install
+define Package/gluon-mesh-batman-adv-common/install
 	$(Gluon/Build/Install)
-
-	$(INSTALL_DIR) $(1)/lib/gluon/mesh-batman-adv
-	echo 14 > $(1)/lib/gluon/mesh-batman-adv/compat
```

Comment:
If you remove the compat file, you also need to adjust the respondd provider that reads it.